spec/var-op-test.test.sh

```diff
@@ -540,3 +540,154 @@ echo "${undef-\e}"
 ## END
 ## BUG zsh/mksh stdout-repr: '\x1b\n'
 ## BUG yash stdout: e
+
+
+#### op-test for ${a[@]} and ${a[*]}
+case $SH in dash) exit ;; esac
+
+test-hyphen() {
+  echo "a   : '${a-no-colon}' '${a:-with-colon}'"
+  echo "a[0]: '${a[0]-no-colon}' '${a[0]:-with-colon}'"
+  echo "a[@]: '${a[@]-no-colon}' '${a[@]:-with-colon}'"
+  echo "a[*]: '${a[*]-no-colon}' '${a[*]:-with-colon}'"
+}
+
+a=()
+test-hyphen
+a=("")
+test-hyphen
+a=("" "")
+test-hyphen
+IFS=
+test-hyphen
+
+## STDOUT:
+a   : 'no-colon' 'with-colon'
+a[0]: 'no-colon' 'with-colon'
+a[@]: 'no-colon' 'with-colon'
+a[*]: 'no-colon' 'with-colon'
+a   : '' 'with-colon'
+a[0]: '' 'with-colon'
+a[@]: '' 'with-colon'
+a[*]: '' 'with-colon'
+a   : '' 'with-colon'
+a[0]: '' 'with-colon'
+a[@]: ' ' ' '
+a[*]: ' ' ' '
+a   : '' 'with-colon'
+a[0]: '' 'with-colon'
+a[@]: ' ' ' '
+a[*]: '' 'with-colon'
+## END
+
+# Bash 2.0..4.4 has a bug that "${a[@]:-xxx}" produces an empty string.  It
+# seemed to consider a[@] and a[*] are non-empty when there is at least one
+# element even if the element is empty.  This was fixed in Bash 5.0.
+#
+# ## BUG bash STDOUT:
+# a   : 'no-colon' 'with-colon'
+# a[0]: 'no-colon' 'with-colon'
+# a[@]: 'no-colon' 'with-colon'
+# a[*]: 'no-colon' 'with-colon'
```

Comment:
I think it could be useful to split this test up like the other one, since bash agrees with OSH on the first part

Although it is a bit annoying because you have to share `test-hyphen`

I suppose that could be `source spec/testdata/var-op-test-helper.sh` or something

Or I guess it is OK to copy and paste too, since the function is short

Reply:
361ac8ca3